tests/ui/hashset_insert_after_contains.rs

```diff
@@ -0,0 +1,77 @@
+#![allow(unused)]
+#![allow(clippy::nonminimal_bool)]
+#![allow(clippy::needless_borrow)]
+#![warn(clippy::hashset_insert_after_contains)]
+
+use std::collections::HashSet;
+
+fn main() {
+    should_warn_cases();
+
+    should_not_warn_cases();
+}
+
+fn should_warn_cases() {
+    let mut set = HashSet::new();
+    let value = 5;
+
+    if !set.contains(&value) {
+        set.insert(value);
+        println!("Just a comment");
+    }
+
+    if set.contains(&value) {
+        set.insert(value);
+        println!("Just a comment");
+    }
+
+    if !set.contains(&value) {
+        set.insert(value);
+    }
+
+    if !!set.contains(&value) {
+        set.insert(value);
+        println!("Just a comment");
+    }
+
+    if (&set).contains(&value) {
+        set.insert(value);
+    }
+
+    let borrow_value = &6;
+    if !set.contains(borrow_value) {
+        set.insert(*borrow_value);
+    }
+
+    let borrow_set = &mut set;
+    if !borrow_set.contains(&value) {
+        borrow_set.insert(value);
+    }
```

Comment:
How about

```rust
if set.contains(&value) {
    println!("value is already in set");
} else {
    set.insert(value);
}
```

Reply:
Good point. Right now the code is only searching for the `insert` in the `then` part of the IF. I think it would make sense to search as well in the `else` part. I am just not sure if we would not warn if we find it in the `else` and the `then` has "a lot of things" (meaning that probably it was a stylist choice of the developer) or if we keep it simple and just warn if we find the `insert` in the else as well... I think that keeping it simple would be the best choice, but, for sure, I am open to suggestions.